Firestore/core/src/firebase/firestore/immutable/array_sorted_map.h

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_IMMUTABLE_ARRAY_SORTED_MAP_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_IMMUTABLE_ARRAY_SORTED_MAP_H_
+
+#include <assert.h>
```

Comment:
Oh, sorry. I wanted to say "consider <cassert> instead". The difference in this case is pretty much pedantic, because that header consists of macros, so it's an entirely optional suggestion.

Technically, all <foo.h> C headers are deprecated. Reading about this now, I found out there's a proposal to un-deprecate <foo.h> headers in C++20 (linked from here https://stackoverflow.com/questions/43011737/is-it-better-to-include-cassert-or-assert-h), so feel free to ignore.

Reply:
It looks like Github's parser eats angle-bracketed words - I wanted to say "consider `<cassert>` instead". The difference in this case is pretty much pedantic, because that header consists of macros, so it's an entirely optional suggestion.

Technically, all <foo.h> C headers are deprecated. Reading about this now, I found out there's a proposal to un-deprecate <foo.h> headers in C++20 (linked from here https://stackoverflow.com/questions/43011737/is-it-better-to-include-cassert-or-assert-h), so feel free to ignore.

Reply:
In general I find these headers to be more trouble than they're worth and I don't know why they were ever created. Interoperability with C is important and these headers make some of the lowest level parts of the system non-interoperable.

I've changed this because #734 greatly reduces the chances we ever need to interoperate with C.